gh repo set-default produces panic: runtime error: index out of range [-1]

### Describe the bug

Trying to use `gh repo set-default` produce a panic runtime error.

```
gh version 2.21.2 (2023-01-03)
```

### Steps to reproduce the behavior

1. Type this `gh repo set-default`
2. Select a default and press enter.
3. See error

### Expected vs actual behavior

Success.

### Logs


```
ssbarnea@m1: ~/c/a/devtools fix/auto-approve
$ gh repo set-default
This command sets the default remote repository to use when querying the
GitHub API for the locally cloned repository.

gh uses the default repository for things like:

 - viewing and creating pull requests
 - viewing and creating issues
 - viewing and creating releases
 - working with Actions
 - adding repository and environment secrets

? Which repository should be the default? ansible-community/devtools
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/cli/cli/v2/pkg/cmd/repo/setdefault.setDefaultRun(0x14000050c30)
	github.com/cli/cli/v2/pkg/cmd/repo/setdefault/setdefault.go:210 +0xae0
github.com/cli/cli/v2/pkg/cmd/repo/setdefault.NewCmdSetDefault.func1(0x140004a2f00?, {0x1033a02b0?, 0x0?, 0x0?})
	github.com/cli/cli/v2/pkg/cmd/repo/setdefault/setdefault.go:103 +0xec
github.com/spf13/cobra.(*Command).execute(0x140004a2f00, {0x1033a02b0, 0x0, 0x0})
	github.com/spf13/cobra@v1.6.1/command.go:916 +0x5c8
github.com/spf13/cobra.(*Command).ExecuteC(0x1400037e600)
	github.com/spf13/cobra@v1.6.1/command.go:1044 +0x35c
main.mainRun()
	github.com/cli/cli/v2/cmd/gh/main.go:212 +0xa9c
main.main()
	github.com/cli/cli/v2/cmd/gh/main.go:49 +0x1c
```

```
# .git/config
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
[remote "origin"]
	url = https://github.com/ansible-community/devtools
	fetch = +refs/heads/*:refs/remotes/origin/*
	gh-resolved = base
[branch "main"]
	remote = origin
	merge = refs/heads/main
[remote "fork"]
	url = https://github.com/ssbarnea/devtools.git
	fetch = +refs/heads/*:refs/remotes/fork/*
[branch "feat/release"]
	remote = fork
	merge = refs/heads/feat/release
[branch "feat/ack"]
	remote = fork
	merge = refs/heads/feat/ack
[branch "fix/0"]
	remote = fork
	merge = refs/heads/fix/0
[branch "fix/1"]
	remote = fork
	merge = refs/heads/fix/1
[branch "fix/skip"]
	remote = fork
	merge = refs/heads/fix/skip
[branch "fix/push-token"]
	remote = fork
	merge = refs/heads/fix/push-token
[branch "fix/refresh-draft"]
	remote = fork
	merge = refs/heads/fix/refresh-draft
[branch "fix/repolist"]
	remote = fork
	merge = refs/heads/fix/repolist
[branch "fix/autolabeler"]
	remote = fork
	merge = refs/heads/fix/autolabeler
[branch "ssbarnea-patch-2"]
	remote = origin
	merge = refs/heads/ssbarnea-patch-2
[branch "feat/matrix"]
	remote = origin
	merge = refs/heads/feat/matrix
[branch "fix/release-drafter"]
	remote = fork
	merge = refs/heads/fix/release-drafter
[branch "chore/more"]
	remote = fork
	merge = refs/heads/chore/more
[branch "fix/auto-approve"]
	remote = fork
	merge = refs/heads/fix/logos
```

--- a/internal/prompter/prompter.go
+++ b/internal/prompter/prompter.go
@@ -57,7 +57,14 @@ func (p *surveyPrompter) Select(message, defaultValue string, options []string) 
 	}
 
 	if defaultValue != "" {
-		q.Default = defaultValue
+		// in some situations, defaultValue ends up not being a valid option; do
+		// not set default in that case as it will make survey panic
+		for _, o := range options {
+			if o == defaultValue {
+				q.Default = defaultValue
+				break
+			}
+		}
 	}
 
 	err = p.ask(q, &result)

